### PR TITLE
Changes related to PENNANT

### PR DESCRIPTION
**Pull request by fkeceli**
1. Added frequency sweep
2. Made number of used cores depending on the system
3. Fixed geopm markup so it is not creating too many epochs per sec.

### CODE REVIEW

File: integration/apps/pennant/pennant.py [hunk 1]
```diff
@@ -43,21 +43,27 @@ def setup_run_args(parser):
...
                         action='store', type=str,
                         default="PENNANT/test/leblancx4/leblancx4.pnt",
                         help='Path to the input file (see .pnt files in test' +
                              'directory of the PENNANT source tarball). ' +
                              'Absolute path or relative to the app ' +
-                             'directory. Default is nohpoly.pnt')
+                             'directory. Default is leblancx4.pnt')
+    parser.add_argument('--cores-per-node', '-c', dest='cores_per_node',
+                        action='store', type=int,
+                        help='Number of pyhysical cores to reserve for the app. ' +
```
Comment (dianarg) [inline integration/apps/pennant/pennant.py]: physical

File: integration/apps/pennant/pennant.py [hunk 2]
```diff
@@ -43,21 +43,27 @@ def setup_run_args(parser):
     """ Add common arguments for all run scripts:
```
Comment (dianarg) [inline integration/apps/pennant/pennant.py:43]: I like this pattern of having a setup_run_args for the app.  We do need to be careful of collisions with the base scripts.  Perhaps a prefix like `--pennant-input`  and `--pennant-cores-per-node`?

File: integration/experiment/frequency_sweep/run_frequency_sweep_pennant.py [hunk 3]
```diff
@@ -0,0 +1,55 @@
...
+from experiment import machine
+from apps.pennant import pennant
+
+if __name__ == '__main__':
+
+    parser = argparse.ArgumentParser()
+    frequency_sweep.setup_run_args(parser)
+    pennant.setup_run_args(parser)
+    args, extra_args = parser.parse_known_args()
+    if len(extra_args) > 0:
```
Comment (dianarg) [inline integration/experiment/frequency_sweep/run_frequency_sweep_pennant.py]: extra_args could be used to inject other launcher arguments like `--geopm-ctl` or options for srun.

File: integration/apps/pennant/pennant.py (hunk 2 above)
Comment (fkeceli) [reply integration/apps/pennant/pennant.py:43]: It was @bgeltz s idea :) I like it too. That's a good idea, I might have to make that change for other apps as well.

Comment (fkeceli) [reply integration/apps/pennant/pennant.py:43]: Although, if there were conflicts argparse would crash anyway but I get the point...

File: integration/experiment/frequency_sweep/run_frequency_sweep_pennant.py (hunk 3 above)
Comment (fkeceli) [reply integration/experiment/frequency_sweep/run_frequency_sweep_pennant.py]: Got it.

File: integration/apps/pennant/pennant.py [hunk 4]
```diff
@@ -43,21 +43,27 @@ def setup_run_args(parser):
...
         --input
     """
-    parser.add_argument('--input', "-i",
+    parser.add_argument('--pennant-input', '-i',
```
Comment (dianarg) [inline integration/apps/pennant/pennant.py]: this needs `dest='pennant_input'`

File: integration/apps/pennant/pennant.py [hunk 5]
```diff
@@ -43,21 +43,27 @@ def setup_run_args(parser):
...
                         action='store', type=str,
                         default="PENNANT/test/leblancx4/leblancx4.pnt",
                         help='Path to the input file (see .pnt files in test' +
                              'directory of the PENNANT source tarball). ' +
                              'Absolute path or relative to the app ' +
-                             'directory. Default is nohpoly.pnt')
+                             'directory. Default is leblancx4.pnt')
+    parser.add_argument('--pennant-cores-per-node', '-c', dest='cores_per_node',
```
Comment (dianarg) [inline integration/apps/pennant/pennant.py]: dest='pennant_cores_per_nodes'

File: integration/apps/pennant/pennant.py [hunk 6]
```diff
@@ -66,13 +72,25 @@ def __init__(self, problem_file="PENNANT/test/leblancx4/leblancx4.pnt"):
...
         else:
             raise RuntimeError("Input file not found: " + problem_file)
+        if cores_per_node is None:
+            # Reserve one for geopm.
+            reserved_cores = mach.num_core() - 1
+            # If odd number throw one out.
+            reserved_cores = (reserved_cores // 2) * 2
+            self._cores_per_node = reserved_cores
+        else:
+            self._cores_per_node = cores_per_node
```
Comment (dianarg) [inline integration/apps/pennant/pennant.py:82]: might want to check that this is <= num_core()

**Review by dianarg — APPROVED** [review]

**Pull request merged**